runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -3664,6 +3664,151 @@ TR::Register *J9::X86::TreeEvaluator::longBitCount(TR::Node *node, TR::CodeGener
    return resultReg;
    }
 
+inline void generateInlinedCheckCastForDynamicCastClass(TR::Node* node, TR::CodeGenerator* cg)
+   {
+   auto use64BitClasses = TR::Compiler->target.is64Bit() &&
+               (!TR::Compiler->om.generateCompressedObjectHeaders() ||
+               (cg->comp()->compileRelocatableCode() && cg->comp()->getOption(TR_UseSymbolValidationManager)));
+   TR::Register *ObjReg = cg->evaluate(node->getFirstChild());
+   TR::Register *castClassReg = cg->evaluate(node->getSecondChild());
+   TR::Register *temp1Reg = cg->allocateRegister();
+   TR::Register *temp2Reg = cg->allocateRegister();
+   TR::Register *temp3Reg = cg->allocateRegister();
```

Comment:
`temp3Reg` seems always holds the `J9Class` of `ObjReg`. It may be better to reflect this from its name to improve readability.
`temp1Reg` and `temp2Reg` are legit temporary registers. I agree to keep the name; however it may be easier to read the code if there are some comments describing their content wherever used, OR aliasing the variable with a meaning name before using it.

Reply:
Addressed. Renamed temp3Reg and added more verbose comments when temp1/2 is (re)defined.